kore/src/Kore/Step/Simplification/TermLike.hs

```diff
@@ -477,3 +496,149 @@ ensureSimplifiedCondition repr termLike condition
             , Pretty.indent 2 "while simplifying:"
             , Pretty.indent 4 (unparse termLike)
             ]
+
+-- | Simplify the given 'TermLike'. Do not simplify any side conditions.
+simplifyOnly ::
```

Comment:
So, is the plan to replace the old term simplifier with this function?

Reply:
Yes, we should eventually replace the old `TermLike` simplifier with this one. There are still some problems with the particular delegates of this function. For example, they are mutually recursive, instead of letting the loop in `simplifyOnly` re-simplify patterns. Also, they don't always delegate to the `Predicate` simplifier like they should.